### PR TITLE
fix bug #5624

### PR DESCRIPTION
**Pull request by hoangviet1985**
fix bug #5624, tested with the example on #5034

### CODE REVIEW

File: modules/core/test/ocl/test_channels.cpp [hunk 1]
```diff
@@ -213,7 +214,7 @@ OCL_TEST_P(Split, Accuracy)
     for (int j = 0; j < test_loop_times; j++)
     {
         generateTestData();
-
+std::cout<<"Hello!"<<std::endl;
```
Comment (PhilLab) [inline modules/core/test/ocl/test_channels.cpp]: This line seems to be committed by accident

**Comment by hoangviet1985** [comment]
Thank you

File: modules/core/src/convert.cpp [hunk 2]
```diff
@@ -924,15 +924,29 @@ void cv::split(InputArray _m, OutputArrayOfArrays _mv)
...
 
+    /*
+    Size size = m.size();
+    int depth = m.depth(), cn = m.channels();
+    _mv.create(cn, 1, depth);
+    for (int i = 0; i < cn; ++i)
+        _mv.create(size, depth, i);
+    std::vector<Mat> dst;
+    _mv.getMatVector(dst);
+     */
```
Comment (mshabunin) [inline modules/core/src/convert.cpp]: Please remove commented code if it is not needed

File: modules/core/test/ocl/test_channels.cpp [hunk 3]
```diff
@@ -213,7 +214,6 @@ OCL_TEST_P(Split, Accuracy)
     for (int j = 0; j < test_loop_times; j++)
     {
         generateTestData();
-
```
Comment (mshabunin) [inline modules/core/test/ocl/test_channels.cpp]: I think this file can be reverted

File: modules/core/src/mathfuncs.cpp [hunk 4]
```diff
@@ -2088,7 +2088,7 @@ double cv::solvePoly( InputArray _coeffs0, OutputArray _roots0, int maxIters )
             for( j = 0; j < n; j++ )
             {
                 num = num*p + coeffs[n-j-1];
-                if( j != i ) denom = denom * (p - roots[j]);
+                if( j != i ) {denom = denom * (p - roots[j]);}
```
Comment (mshabunin) [inline modules/core/src/mathfuncs.cpp]: This file can be reverted to avoid merging conflict with other PRs.

File: modules/core/src/convert.cpp [hunk 5]
```diff
@@ -924,11 +924,16 @@ void cv::split(InputArray _m, OutputArrayOfArrays _mv)
...
+    buffer.allocate(m.dims);
+    int* size = (int*) buffer;
+    for(int i = 0; i < m.dims; i++)
+        *(size + i) = m.size[i];
+
     int depth = m.depth(), cn = m.channels();
     _mv.create(cn, 1, depth);
     for (int i = 0; i < cn; ++i)
-        _mv.create(size, depth, i);
+        _mv.create(m.dims, size, depth, i);
```
Comment (mshabunin) [inline modules/core/src/convert.cpp]: Can `m.size.p` be used here instead of `size`?

Comment (hoangviet1985) [inline modules/core/src/convert.cpp]: you are right! That makes the code shorter. Thank you

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**